src/tribler-core/tribler_core/modules/metadata_store/community/discovery_booster.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from ipv8.peerdiscovery.discovery import EdgeWalk
+
+
+class DiscoveryBooster:
+    def __init__(self, timeout_in_sec=10.0, max_peers=200, take_step_interval_in_sec=0.05):
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+        self.timeout_in_sec = timeout_in_sec
+        self.max_peers = max_peers
+        self.take_step_interval_in_sec = take_step_interval_in_sec
+
+        self.community = None
+        self.saved_max_peers = None
+        self.walker = None
+
+        self._take_step_task_name = 'take step'
+
+    def apply(self, community):
+        if not community:
+            return
+
+        self.logger.info(
+            f'Apply. Timeout: {self.timeout_in_sec}s, '
+            f'Max peers: {self.max_peers}, '
+            f'Take step interval: {self.take_step_interval_in_sec}s'
+        )
+
+        self.community = community
+        self.saved_max_peers = community.max_peers
+
+        community.max_peers = self.max_peers
+
+        # values for neighborhood_size and edge_length were found empirically to
+        # maximize peer count at the end of a 30 seconds period
+        self.walker = EdgeWalk(community, neighborhood_size=25, edge_length=25)
+
+        community.register_task(self._take_step_task_name, self.take_step, interval=self.take_step_interval_in_sec)
+
+        community.register_anonymous_task('', self.finish, delay=self.timeout_in_sec)
```

Comment:
It's good practice to give your anonymous tasks a basename that is not the empty string (in your case perhaps "DiscoveryBooster timeout").

This makes any potentially failing tasks be logged as "\<basename> \<random number>" instead of " \<random number>", allowing for easier debugging.

Reply:
👌 thank you for the explanation.